src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateItemEventHandler.java

```diff
@@ -0,0 +1,162 @@
+        .setContent(new JsonObject((LinkedHashMap) dataImportEventPayload.getCurrentNode().getContent()).mapTo(MappingProfile.class));
+
+      dataImportEventPayload.getContext().put(ITEM.value(), new JsonObject().encode());
+
+      MappingManager.map(dataImportEventPayload);
+      JsonObject itemAsJson = new JsonObject(dataImportEventPayload.getContext().get(ITEM.value()));
+      itemAsJson.put("id", UUID.randomUUID().toString());
+
+      ItemCollection itemCollection = storage.getItemCollection(context);
+      List<String> errors = validateItem(itemAsJson);
+      if (errors.isEmpty()) {
+        Item mappedItem = ItemUtil.jsonToItem(itemAsJson);
+        isItemBarcodeUnique(itemAsJson.getString("barcode"), itemCollection)
+          .compose(isUnique -> isUnique
+            ? addItem(mappedItem, itemCollection)
+            : Future.failedFuture(String.format("Barcode must be unique, %s is already assigned to another item", itemAsJson.getString("barcode"))))
+          .setHandler(ar -> {
+            if (ar.succeeded()) {
+              dataImportEventPayload.getContext().put(ITEM.value(), itemAsJson.encode());
+              dataImportEventPayload.setEventType(ITEM_CREATED_EVENT_TYPE);
+              future.complete(dataImportEventPayload);
+            } else {
+              LOG.error("Error creating inventory Item" , ar.cause());
+              future.completeExceptionally(ar.cause());
+            }
+          });
+      } else {
+        String msg = String.format("Mapped Item is invalid: %s", errors.toString());
+        LOG.error(msg);
+        future.completeExceptionally(new EventProcessingException(msg));
+      }
+    } catch (IOException e) {
+      LOG.error("Error creating inventory Item" , e);
+      future.completeExceptionally(e);
+    }
+    return future;
+  }
+
+  @Override
+  public boolean isEligible(DataImportEventPayload dataImportEventPayload) {
+    if (dataImportEventPayload.getCurrentNode() != null && ACTION_PROFILE == dataImportEventPayload.getCurrentNode().getContentType()) {
+      ActionProfile actionProfile = JsonObject.mapFrom(dataImportEventPayload.getCurrentNode().getContent()).mapTo(ActionProfile.class);
+      return actionProfile.getAction() == CREATE && actionProfile.getFolioRecord() == ITEM;
+    }
+    return false;
+  }
+
+  private List<String> validateItem(JsonObject itemAsJson) {
+    ArrayList<String> errorMessages = new ArrayList<>();
+    for (String fieldPath : requiredFields) {
+      String field = StringUtils.substringBefore(fieldPath, ".");
+      String nestedField = StringUtils.substringAfter(fieldPath, ".");
+      if (!isExistsRequiredProperty(itemAsJson, field, nestedField)) {
+        errorMessages.add(String.format("Field '%s' is a required field and can not be null", fieldPath));
+      }
+    }
+    String statusName = JsonHelper.getNestedProperty(itemAsJson, "status", "name");
+    if (StringUtils.isNotBlank(statusName) && !ItemStatusName.isStatusCorrect(statusName)) {
```

Comment:
Status is also required field, so `status == null` is not allowed.

Reply:
if nested field ` name != null ` then ` status != null ` as well, therefore Item with status == null will not be created

Reply:
Yep, sorry, I meant that `status.name == null` is also unacceptable. See https://github.com/folio-org/mod-inventory-storage/blob/master/ramls/item.json#L245.